### PR TITLE
API documentation improvements

### PR DESCRIPTION
**Pull request by aarzilli**
I've adapted the documentation we had in `service/client.go` for `service/rpc2/server.go` so that godoc generates the API documentation for us. I've also added to `Documentation/api/json-rpc/README.md` a brief explanation of how to use the documentation on godoc.org for our json-rpc service.

### CODE REVIEW

File: Documentation/api/json-rpc/README.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+The first step will be calling the method `FindLocation` with `Scope = api.EvalScope{ GoroutineID: -1, Frame: 0}` and `Loc = "main.main"`. The JSON-RPC request packet should look like this:
+
+```
+{"method":"RPCServer.FindLocation","params":[{"Scope":{"GoroutineID":-1,"Frame":0},"Loc":"main.main"}],"id":2}
+```
+
+the response packet will look like this:
+
+```
+"id":2,"result":{"Locations":[{"pc":4199019,"file":"/home/a/temp/callme/callme.go","line":31,"function":{"name":"main.main","value":4198992,"type":84,"goType":0}}]},"error":null}
```
Comment (derekparker) [inline Documentation/api/json-rpc/README.md]: Looks like you're missing a `{` at the beginning here.

File: service/rpc2/server.go [hunk 2]
```diff
@@ -107,6 +107,7 @@ type ProcessPidOut struct {
 	Pid int
 }
 
+// Returns the pid of the process we are debugging.
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `ProcessPid`.

File: service/rpc2/server.go [hunk 3]
```diff
@@ -129,6 +131,7 @@ type RestartIn struct {
 type RestartOut struct {
 }
 
+// Restarts program.
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `Restart`.

File: service/rpc2/server.go [hunk 4]
```diff
@@ -201,6 +207,10 @@ type StacktraceOut struct {
 	Locations []api.Stackframe
 }
 
+// Returns stacktrace of goroutine Id up to the specified Depth.
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `Stacktrace`.

File: service/rpc2/server.go [hunk 5]
```diff
@@ -276,6 +301,10 @@ type AmendBreakpointIn struct {
 type AmendBreakpointOut struct {
 }
 
+// Allows user to update an existing breakpoint for example to change the information
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `AmendBreakpoint`.

File: service/rpc2/server.go [hunk 6]
```diff
@@ -499,6 +544,7 @@ type AttachedToExistingProcessOut struct {
 	Answer bool
 }
 
+// Returns whether we attached to a running process or not
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `AttachedToExistingProcess`.

File: service/rpc2/server.go [hunk 7]
```diff
@@ -515,6 +561,19 @@ type FindLocationOut struct {
 	Locations []api.Location
 }
 
+// Returns concrete location information described by a location expression
```
Comment (derekparker) [inline service/rpc2/server.go]: nit: Start comment with `FindLocation`.

**Comment by aarzilli** [comment]
done.

**Pull request merged**